clientd3d/d3drender_objects.c

```diff
@@ -0,0 +1,2408 @@
+// Meridian 59, Copyright 1994-2024 Andrew Kirmse and Chris Kirmse.
+// All rights reserved.
+//
+// This software is distributed under a license that is described in
+// the LICENSE file that accompanies it.
+//
+// Meridian is a registered trademark.
+#include "client.h"
+#include <unordered_map>
+#include <unordered_set>
+#include <chrono>
+
+// Variables
+
+extern LPDIRECT3DVERTEXDECLARATION9 decl1dc;
+extern LPDIRECT3DVERTEXDECLARATION9 decl2dc;
+extern PDIRECT3DTEXTURE9 gpBackBufferTexFull;
+extern LPDIRECT3DTEXTURE9 gpBackBufferTex[16];
+
+extern d3d_render_pool_new gObjectPool;
+extern d3d_render_cache_system gObjectCacheSystem;
+extern d3d_driver_profile gD3DDriverProfile;
+
+// Main client windows current viewport area
+extern int main_viewport_width;
+extern int main_viewport_height;
+
+extern font_3d gFont;
+
+extern Color base_palette[];
+
+extern int gSmallTextureSize;
+extern int sector_depths[];
+extern long nitems;
+extern DrawItem drawdata[];
+extern ObjectRange visible_objects[];    /* Where objects are on screen */
+extern int num_visible_objects;
+extern int gNumObjects;
+
+extern d_light_cache gDLightCache;
+extern d_light_cache gDLightCacheDynamic;
+
+extern long timeObjects;
+
+static auto TRANSLUCENT_FLAGS = OF_TRANSLUCENT25 | OF_TRANSLUCENT50 | OF_TRANSLUCENT75 | OF_DITHERTRANS;
+
+extern player_info player;
+extern RECT gD3DRect;
+
+static D3DMATRIX mat, trans, view, proj, identity;
```

Comment:
I think you can actually delete some or all of these?  They tend to be declared as local variables and don't need to exist past the scope of a function.